src/Rings/mpoly-ideals.jl

```diff
@@ -572,7 +571,31 @@ Ideal generated by
   3*a*b*c
 ```
 """
-@attr T function radical(I::T) where {T <: MPolyIdeal}
+function radical(
+    I::MPolyIdeal{T}; 
+    eliminate_variables::Bool=true
+  ) where {T <: MPolyRingElem}
+  get_attribute(I, :radical) do
+    if eliminate_variables
+      is_known_to_be_radical(I) && return I
+      # Calling `elimpart` (within `simplify`) turns out to significantly speed things up in many cases.
+      R = base_ring(I)
+      Q, pr = quo(R, I)
+      S, iso, iso_inv = simplify(Q)
+      is_zero(ngens(S)) && return I # This only happens when all variables can be eliminated. 
+      # Then the ideal defines a reduced point.
+      J = modulus(S)
+      pre_res = _compute_radical(J)
+      pre_res = ideal(R, elem_type(R)[lift(iso_inv(S(g))) for g in gens(pre_res)])
+      res = ideal(R, small_generating_set(pre_res + I))
+      set_attribute!(res, :is_radical=>true)
+      return res
+    end
```

Comment:
As you specifically asked my opinion on this code block: Yes, this definitely makes sense, as it prepares the objects better for the use of Singular's primdec.lib.